### DISCUSSION TITLE
storage usage is increasing when milvus cluster is idle

### DISCUSSION THREAD

**Discussion by LiuAlex1109** [milvus-io/milvus#32715] [Q&A and General discussion] [answered]
The pulsar still occupied disk room even if I delete all collection in milvus, I want to know how can i release these room?
```
[root@localhost nfs_data]# du -h ./ --max-depth=1
212K./default-milvus-minio-pvc-d4f2ed31-3b6d-4635-bf0e-c04dc3cae047
700K./default-data-milvus-etcd-0-pvc-164b44cf-603a-4113-989b-824ca1292645
316K./default-milvus-pulsar-zookeeper-data-milvus-pulsar-zookeeper-0-pvc-1c9a5d19-ba2e-41f2-9a33-e2a5e07c16be
79M./default-milvus-pulsar-bookie-journal-milvus-pulsar-bookie-1-pvc-014618af-2ccd-4766-9e4e-ea4b564f3344
2.7M./default-milvus-pulsar-bookie-ledgers-milvus-pulsar-bookie-0-pvc-43b9fbb5-ca30-4614-b3f8-d5906200b941
79M./default-milvus-pulsar-bookie-journal-milvus-pulsar-bookie-0-pvc-84e20d03-3374-4fb4-98c8-88fdbe197879
2.7M./default-milvus-pulsar-bookie-ledgers-milvus-pulsar-bookie-1-pvc-12e69864-fd86-48ac-8b7c-6206ec806a79
164M./
```

**Comment by yhmo** [comment, marked as answer]
Pulsar is like a WAL(write-ahead-log) component for Milvus. All the data is first stored in Pulsar and then consumed by Milvus.
Pulsar is an independent service, maintains its messages by itself.

The "defaultRetentionTimeInMinutes" controls the retention time policy of Pulsar, which is mentioned here: https://pulsar.apache.org/docs/next/cookbooks-retention-expiry/

In Milvus helm, it is defined here: https://github.com/zilliztech/milvus-helm/blob/ce3fbd0dfa7d5393ed93db3416eae136c0c92c69/charts/milvus/values.yaml#L752

The default value is 7 days, you can change it to a small value. We highly recommend setting the value to be larger than 1 day.

**Reply by LiuAlex1109** [reply]
Got it! Thank you for your apply.
